prog/vnet/aws/vpc_nexus.rb

```diff
@@ -0,0 +1,193 @@
+        allow_ingress(security_group_response.group_id, firewall_rule.port_range.first, firewall_rule.port_range.last - 1, firewall_rule.cidr.to_s)
+      end
+
+      # Allow SSH ingress from the internet so that the controlplane can verify
+      # that the VM is running.
+      allow_ingress(security_group_response.group_id, 22, 22, "0.0.0.0/0")
+      hop_create_route_table
+    end
+    nap 1
+  end
+
+  label def create_route_table
+    # Step 3: Update the route table for ipv_6 traffic
+    route_table_response = client.describe_route_tables({filters: [{name: "vpc-id", values: [private_subnet.private_subnet_aws_resource.vpc_id]}]})
+    route_table_id = route_table_response.route_tables[0].route_table_id
+    private_subnet.private_subnet_aws_resource.update(route_table_id: route_table_id)
+    internet_gateway_response = client.describe_internet_gateways({filters: [{name: "tag:Name", values: [private_subnet.name]}]})
+
+    if internet_gateway_response.internet_gateways.empty?
+      internet_gateway_id = client.create_internet_gateway({
+        tag_specifications: Util.aws_tag_specifications("internet-gateway", private_subnet.name)
+      }).internet_gateway.internet_gateway_id
+      private_subnet.private_subnet_aws_resource.update(internet_gateway_id:)
+      client.attach_internet_gateway({internet_gateway_id:, vpc_id: private_subnet.private_subnet_aws_resource.vpc_id})
+    else
+      internet_gateway = internet_gateway_response.internet_gateways.first
+      internet_gateway_id = internet_gateway.internet_gateway_id
+      private_subnet.private_subnet_aws_resource.update(internet_gateway_id:)
+      if internet_gateway.attachments.empty?
+        client.attach_internet_gateway({internet_gateway_id:, vpc_id: private_subnet.private_subnet_aws_resource.vpc_id})
+      end
+    end
+
+    begin
+      client.create_route({
+        route_table_id: route_table_id,
+        destination_ipv_6_cidr_block: "::/0",
+        gateway_id: internet_gateway_id
+      })
+
+      client.create_route({
+        route_table_id: route_table_id,
+        destination_cidr_block: "0.0.0.0/0",
+        gateway_id: internet_gateway_id
+      })
+    rescue Aws::EC2::Errors::RouteAlreadyExists
+    end
+
+    hop_wait
+  end
+
+  label def wait
+    when_update_firewall_rules_set? do
+      private_subnet.vms.each(&:incr_update_firewall_rules)
+      decr_update_firewall_rules
+    end
+
+    private_subnet.semaphores.each(&:destroy)
+
+    nap 60 * 60 * 24 * 365
+  end
+
+  label def destroy
+    if private_subnet.nics.any? { |n| !n.vm_id.nil? }
+      register_deadline(nil, 10 * 60, allow_extension: true) if private_subnet.nics.any? { |n| n.vm&.prevent_destroy_set? }
+
+      Clog.emit("Cannot destroy subnet with active nics, first clean up the attached resources") { private_subnet }
+
+      nap 5
+    end
+
+    decr_destroy
+    private_subnet.nics.each(&:incr_destroy)
+    private_subnet.firewalls.each(&:destroy)
```

Comment:
This looks wrong (maybe it isn't, but it certainly looks like it is). The firewalls could be used by other private subnets. If you want to remove the entries from the join table:


```suggestion
    private_subnet.remove_all_firewalls
```

Reply:
This was copy/paste from the earlier prog, but I think you are right. `remove_all_firewalls` makes more sense.

cc @furkansahin to confirm whether this is OK or if we are missing something.